From ee2033e9c64139c4f052bed52e72eba44a08b40a Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Tue, 22 Jun 2021 13:26:24 +0900
Subject: [PATCH] ksmbd: change stream type macro to enumeration
Git-commit: ee2033e9c64139c4f052bed52e72eba44a08b40a
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Change stream type macro to enumeration and move it to vfs.h.

Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/glob.h | 3 ---
 fs/cifsd/vfs.h  | 8 ++++++++
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/fs/cifsd/glob.h b/fs/cifsd/glob.h
index 8119cb7ddbed..49a5a3afa118 100644
--- a/fs/cifsd/glob.h
+++ b/fs/cifsd/glob.h
@@ -16,9 +16,6 @@
 
 extern int ksmbd_debug_types;
 
-#define DATA_STREAM	1
-#define DIR_STREAM	2
-
 #define KSMBD_DEBUG_SMB		BIT(0)
 #define KSMBD_DEBUG_AUTH	BIT(1)
 #define KSMBD_DEBUG_VFS		BIT(2)
diff --git a/fs/cifsd/vfs.h b/fs/cifsd/vfs.h
index 29352c227028..a9c14c5dee8d 100644
--- a/fs/cifsd/vfs.h
+++ b/fs/cifsd/vfs.h
@@ -101,6 +101,14 @@ struct xattr_ntacl {
 #define XATTR_NAME_SD_LEN	\
 		(sizeof(XATTR_SECURITY_PREFIX SD_PREFIX) - 1)
 
+/*
+ * Enumeration for stream type.
+ */
+enum {
+	DATA_STREAM	= 1,	/* type $DATA */
+	DIR_STREAM		/* type $INDEX_ALLOCATION */
+};
+
 /* CreateOptions */
 /* Flag is set, it must not be a file , valid for directory only */
 #define FILE_DIRECTORY_FILE_LE			cpu_to_le32(0x00000001)
-- 
2.33.0

